applications/MeshMovingApplication/custom_utilities/parametric_linear_transform.h

```diff
@@ -0,0 +1,170 @@
+                              const Parameters& rReferencePoint,
+                              const Parameters& rTranslationVector);
+
+    /** Construct via euler angles
+     *  @param rEulerAngles euler angles (radians, array of size 3)
+     *  @param rReferencePoint origin of rotation (array of size 3)
+     *  @param rTranslationVector translation vector (array of size 3)
+     *  @note The euler angles follow the convention specified by @ref{Quaternion} (Z, -X', Z")
+     */
+    ParametricLinearTransform(const Parameters& rEulerAngles,
+                              const Parameters& rReferencePoint,
+                              const Parameters& rTranslationVector);
+
+    ParametricLinearTransform(const ParametricLinearTransform& rOther) = default;
+
+    ParametricLinearTransform(ParametricLinearTransform&& rOther) = default;
+
+    ParametricLinearTransform& operator=(const ParametricLinearTransform& rOther) = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /** Evaluate the transformation at the given location and apply it to a position vector, returning its transformed version
+     *  @note the transformation is not updated unless at least one transformation-related
+     *  quantity has changed since the last call to ParametricLinearTransform::Apply.
+     *  @param rPoint point to be transformed; it also acts as the current position input to @ref{GenericFunctionUtility}
+     *  @param t time, see @ref{GenericFunctionUtility}
+     *  @param X initial x-coordinate, see @ref{GenericFunctionUtility}
+     *  @param Y initial y-coordinate, see @ref{GenericFunctionUtility}
+     *  @param Z initial z-coordinate, see @ref{GenericFunctionUtility}
+     */
+    array_1d<double,3> Apply(const array_1d<double,3>& rPoint,
+                             const double t,
+                             const double X = 0.0,
+                             const double Y = 0.0,
+                             const double Z = 0.0);
+
+    ///@}
+
+private:
+
+    ///@name Private type definitions
+    ///@{
+
+    // Necessary to parse constants
+    static std::string ExtractFunctionBody(const Parameters& rParameters);
+
+    /// Class for storing and evaluating an array of @ref{GenericFunctionUtility}
+    template <std::size_t ArraySize>
+    class VectorFunction : public std::array<FunctionType::Pointer,ArraySize>
+    {
+    public:
+        VectorFunction(const Parameters& rFunctions)
+        {
+            KRATOS_TRY
+
+            KRATOS_ERROR_IF_NOT(rFunctions.IsArray()) << "VectorFunction can only be constructed from a parameter array";
+
+            // Assign components - bounds checks are performed in Parameters::GetArrayItem
+            for (std::size_t i=0; i<ArraySize; ++i) {
+                const std::string function_string = ExtractFunctionBody(rFunctions.GetArrayItem(i));
+                this->at(i) = std::make_shared<GenericFunctionUtility>(function_string);
```

Comment:
why shared?

Reply:
True, ```unique``` should suffice.

Reply:
On second thought, ```VectorFunction``` needs to be captured in a lambda by value (or move) [here](https://github.com/KratosMultiphysics/Kratos/blob/ef65945f062ab157b4d71de8ecd634fd5aec4341/applications/MeshMovingApplication/custom_utilities/parametric_linear_transform.cpp#L70), but move captures are only supported from C++14 onwards.

https://isocpp.org/wiki/faq/cpp14-language#lambda-captures

Reply:
:cry: